github4s/src/main/scala/github4s/interpreters/RepositoriesInterpreter.scala

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2016-2020 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package github4s.interpreters
+
+import github4s.http.HttpClient
+import github4s.algebras.Repositories
+import cats.data.NonEmptyList
+import github4s.GithubResponses.GHResponse
+import github4s.domain._
+import github4s.Decoders._
+import github4s.Encoders._
+
+class RepositoriesInterpreter[F[_]](implicit client: HttpClient[F], accessToken: Option[String])
+    extends Repositories[F] {
+  override def get(
+      owner: String,
+      repo: String,
+      headers: Map[String, String] = Map()): F[GHResponse[Repository]] =
+    client.get[Repository](accessToken, s"repos/$owner/$repo", headers)
+
+  override def listOrgRepos(
+      org: String,
+      `type`: Option[String],
+      pagination: Option[Pagination],
+      headers: Map[String, String] = Map()): F[GHResponse[List[Repository]]] =
+    client.get[List[Repository]](
+      accessToken,
+      s"orgs/$org/repos",
+      headers,
+      params = `type`.map(t => Map("type" -> t)).getOrElse(Map.empty),
```

Comment:
```suggestion
      params = `type`.fold(Map.empty[String,String])(t => Map("type" -> t))
```
I think maybe each `map` followed by a `getOrElse` can be replaced by `fold`?

Reply:
@angoglez Good catch!